samcli/commands/exceptions.py

```diff
@@ -68,9 +68,9 @@ def show(self, file: Optional[IO] = None) -> None:
         click.secho(msg, file=file, err=True, fg="yellow")
 
 
-class CredentialsError(UserException):
+class AWSServiceClientError(UserException):
```

Comment:
How does this end up looking in our telemetry system? I know we capture errors that are thrown. My concern in changing this to be generic is we loose the ability to understand what errors we are throwing. Capturing "CredentialError" vs "AWSServiceClientError" for missing/invalid creds is vastly different.

Reply:
I can rename it back to CredentialError for the sake of consistency, but ClientError covers a large suite of errors.

Reply:
This was more of a meta point. I am ok either way.